pycbc/inference/models/relbin.py

```diff
@@ -231,15 +230,28 @@ def __init__(
             fpoints = Array(self.f[ifo].astype(numpy.float64))
             fpoints = fpoints[self.kmin[ifo]:self.kmax[ifo]+1]
 
-            if self.no_det_response:
+            if self.apply_det_response:
                 wave = get_fd_det_waveform_sequence(ifos=ifo,
                                                     sample_points=fpoints,
                                                     **self.fid_params)
                 curr_wav = wave[ifo]
+                self.ta[ifo] = 0
             else:
                 fid_hp, fid_hc = get_fd_waveform_sequence(sample_points=fpoints,
                                                           **self.fid_params)
-                curr_wav = fid_hp
+                # Apply detector response if not handled by
+                # the waveform generator
+                self.det[ifo] = Detector(ifo)
+                dt = self.det[ifo].time_delay_from_earth_center(
+                    self.fid_params["ra"],
+                    self.fid_params["dec"],
+                    self.fid_params["tc"],
+                )
+                self.ta[ifo] = self.fid_params["tc"] + dt
+                fp, fc = self.det[ifo].antenna_pattern(
+                    self.fid_params["ra"], self.fid_params["dec"],
+                    self.fid_params["polarization"], self.fid_params["tc"])
+                curr_wav = (fid_hp * fp + fid_hc * fc)
```

Comment:
So this change is a fix for the reference waveforms, when applying detector response ... Seems sane.

Reply:
Ah, no, it just cleans up the code below.